addons/nightvision/XEH_preInit.sqf

```diff
@@ -2,6 +2,23 @@
 
 ADDON = false;
 
+PREP_RECOMPILE_START;
 #include "XEH_PREP.hpp"
+PREP_RECOMPILE_END;
+
+// Credit to killzonekid
+// Almost the same as (([] call CBA_fnc_getFov) select 1) but 10x faster
+DFUNC(kkTrueZoom) = {
+    (
+    [0.5,0.5]
+    distance2D
+    worldToScreen
+    positionCameraToWorld
+    [0,3,4]
+    ) * (
+    getResolution
+    select 5
+    ) / 2
```

Comment:
I'm not sure how any of this would effect a script command that's just taking array values and working with pure maths :stuck_out_tongue_closed_eyes:

It sounds like your problems stem from `getPos` not `distance`